mule-deployer/src/main/java/org/mule/tools/deployment/cloudhub/CloudHubArtifactDeployer.java

```diff
@@ -222,4 +222,15 @@ private Workers getWorkers(Integer amount, String type) {
     workers.setType(workerType);
     return workers;
   }
+
+  protected Map<String, String> resolveProperties(Map<String, String> originalProperties, Map<String, String> properties,
+                                                  Boolean overrideProperties) {
+    if (properties != null) {
+      if (!overrideProperties) {
+        properties.putAll(originalProperties);
+      }
+      originalProperties = properties;
```

Comment:
we should not modify the original properties. I would just `return properties;` here